### DISCUSSION TITLE
Fav List Broken

### DISCUSSION THREAD

**Discussion by UncleFred** [hugoattal/OrbitalMarket#53] [General]
Attempting to import a large favorite list seems to break things. All items vanish from the list. Attempting to import a new list does nothing, neither does clearing cookies
[UnrealMarketFavoriteListOrbital_2024_04.txt](https://github.com/user-attachments/files/16418074/UnrealMarketFavoriteListOrbital_2024_04.txt)

**Comment by hugoattal** [comment]
OH SHIT, I didn't saw that coming...

Those were local IDs of the products in my database, and due to a database problem and a stupid mistake I made, I had to reset it, losing all the IDs correspondence 😭 ...

It's unlikely I'll be able to get those products back, I should have stored the unreal ID instead 😭...

I'm so, so, so sorry, I can't see a way to get your items back... If I can help you in ANY way to construct that list back, tell me! If those were the items from your vault, I'll add a feature to copy-paste the page from your vault (https://www.unrealengine.com/marketplace/en-US/vault) to import it much faster, 100 by 100.

In the meantime, I'll change the save Unreal ID (instead of the local ID), so this problem won't happen in the future.

**Reply by UncleFred** [reply]
Wow! Thanks for letting me know. Most was imported from UnrealSales.io. Luckily, that database is still intact!

I'll manually reconstruct it from there, and then manually go through everything since April.

PS: Please add a favorite heart button within the asset page itself. Right now, if you click on an asset and want to add it to your favorite list, you have to have a separate window open and hunt it down from the asset list search page.

**Reply by hugoattal** [reply]
So, I just shipped a big update

- First, you now have the fav button in the asset itself (near the name)
- You can import assets from the vault by copy/paste the whole page from https://www.unrealengine.com/marketplace/en-US/vault
- You can import assets by separating their name with comma, example:
`Electronic Nodes,Darker Nodes,Ultra Dynamic Sky`
- The import button now uses the clipboard automatically

**Reply by UncleFred** [reply]
There's one major issue I foresee with importing assets based on their name: creators something adjust the asset names. However, I see the URLs don't use an ID number like you'd see on Amazon or Aliexpress, so maybe this is the only option.

**Reply by hugoattal** [reply]
It just matches the name if it exists in the local DB, then it saves the UnrealID associated, so if they change the name in the future, there's no problem.

If you import `Electronic Nodes,Darker Nodes,Ultra Dynamic Sky`
Then, you export `6a1eafd3a6f34a929725585062b403a2,5cb2a394d0c04e73891762be4cbd7216,4b3441f0228a40ec9ca986489a5bd682`

**Reply by UncleFred** [reply]
I rebuilt the list from scratch. If you need a large list to test anything, here it is:
[UnrealMarketFavoriteListOrbital_2024_07.txt](https://github.com/user-attachments/files/16445634/UnrealMarketFavoriteListOrbital_2024_07.txt)
